[docs] CoreDB contributor docs: add multiple remotes section (#28851)
yugabyte/yugabyte-db: docs/content/preview/contribute/core-database/merge-with-upstream-repositories.md

This isn't too big of a problem since most of YugabyteDB's issues on low numbers are already closed, and these subtrees's issue numbers tend to be relatively low.
However, when it does happen, the issues that were inadvertently closed need to manually be reopened.

### Git multiple remotes

When it comes to merging upstream repositories, you will likely have to manage multiple remote repositories.

> This section will illustrate a workflow for that using the PostgreSQL repository as an example.

Reviewer: ```suggestion
This section illustrates a workflow for that using the PostgreSQL repository as an example.
```

First, please get a good understanding of git remotes.

> [This short article](https://dev.to/hashcode01/add-a-second-remote-origin-to-git-35a7) covers the concept well.

Reviewer: ```suggestion
```

> YugabyteDB's [yugabyte/postgres][repo-postgres] repo does not have all branches of upstream PostgreSQL nor is it guaranteed to be synced.

Reviewer: ```suggestion
YugabyteDB's [yugabyte/postgres][repo-postgres] repository does not have all branches of upstream PostgreSQL, nor is it guaranteed to be synced.
```

The common case workflow is to have at least two remotes: one for [yugabyte/postgres][repo-postgres] and one for upstream PostgreSQL.

> The upstream PostgreSQL remote can be set up in one of several ways:

Reviewer: ```suggestion
The upstream PostgreSQL remote can be set up in one of the following ways:
```
